Run CMD on ttyS0 after login and exit the VM after CMD is finished

rmohr opened the pull request:
We somehow have to make sure that SPICE always uses ttyS0

rmohr committed:
Encode xml before writing

fabiand commented on vocker:
```diff
@@ -345,7 +345,7 @@ if __name__ == "__main__":
 
         with tempfile.NamedTemporaryFile("wb") as spec:
-            spec.write(xml)
+            spec.write(xml.encode())
```
I think we need it earlier:

```
@@ -305,7 +309,7 @@ if __name__ == "__main__":
                      "--rng", "/dev/random",
                      "--noautoconsole",
                      "--print-xml")
-        xml = str(xml)
+        xml = str(xml).encode("UTF-8")
```

rmohr replied:
I don't think so. 

```
domroot = ET.fromstring(xml)
```

at line 322 would not work then

rmohr committed:
Run CMD on ttyS0 terminal trough .bashrc

A very simple way to have automatic login with everything set up and
making sure that we run CMD only once by checking in .bashrc if tty
confirms that we run on ttyS0.

After CMD is done, 'shutdown -h now' is invoked.
